model/workload/workload.go

```diff
@@ -53,12 +84,81 @@ func Validate(w *tw.RDTWorkLoad) error {
 	return nil
 }
 
+func setCoreIDAndTaskID(w *tw.RDTWorkLoad, resAss *resctrl.ResAssociation) {
+	if len(w.CoreIDs) >= 0 {
+		bm, _ := rmdbase.CPUBitmaps(w.CoreIDs)
+		oldbm, _ := rmdbase.CPUBitmaps(resAss.CPUs)
+		bm = bm.Or(oldbm)
+		resAss.CPUs = bm.ToString()
+	} else {
+		if len(resAss.CPUs) == 0 {
+			resAss.CPUs = ""
+		}
+	}
+	resAss.Tasks = append(resAss.Tasks, w.TaskIDs...)
+}
+
 // Enforce a user request workload based on defined policy
-func Enforce(w *tw.RDTWorkLoad) *rmderror.AppError {
+func Enforce(w *tw.RDTWorkLoad, mbaInfo *m_mba.Info) *rmderror.AppError {
 	w.Status = tw.Failed
+	var grpName string
+	var resAss *resctrl.ResAssociation
+	newCandidate := make(map[string]mixedCandidate)
+	maxMba := "100"
 
 	l.Lock()
 	defer l.Unlock()
+
+	if w.MaxCache == nil && w.MinCache == nil && w.Policy == "" &&
+		(w.MbaPercentage != nil || w.MbaMbps != nil) {
+		mbaStr := strconv.FormatUint((uint64)(*w.MbaPercentage), 10)
+		cellNum, err := l_mba.GetCellNumber()
+		if err != nil {
+			return rmderror.AppErrorf(http.StatusInternalServerError,
+				"Get cell number failed; %s", err.Error())
+		}
+
+		if len(w.TaskIDs) > 0 {
+			grpName = w.TaskIDs[0] + "-MBA-Task"
+			for i := 0; i < cellNum; i++ {
+				cellID := strconv.FormatInt((int64)(i), 10)
+				fmt.Println(i, cellID)
+				newCandidate[cellID] = mixedCandidate{nil, &mbaStr}
+			}
+		} else if len(w.CoreIDs) > 0 {
+			grpName = w.CoreIDs[0] + "-MBA-Core"
+			for _, idStr := range w.CoreIDs {
+				socketID, err := cpu.LocateOnSocket(idStr)
+				if err != nil {
+					return rmderror.NewAppError(http.StatusInternalServerError,
+						"Error on LocateOnSocket.", err)
+				}
+				if _, ok := newCandidate[socketID]; !ok {
+					newCandidate[socketID] = mixedCandidate{nil, &mbaStr}
+				}
+			}
+			for i := 0; i < cellNum; i++ {
```

Comment:
Shihta, could you explain why you have these lines, if it is for assigning defaults to socket/cell not used, we do not need to explicitly do this.